### PR TITLE
Put all binaries in recalboxFiles.py and make an array of the known b…

### PR DESCRIPTION
**Pull request by substring**
…inaries

That PR is mostly for @supernature2k who is writing a python script to kill a running emulator with a GPIO button.
This way, he has an array of known emulator binaries
And I slightly rewrote the C64 configgen

### CODE REVIEW

File: configgen/recalboxFiles.py [hunk 1]
```diff
@@ -64,7 +69,27 @@
...
+linappleBin = '/usr/bin/linapple'
+
+scummvmBin = '/usr/bin/scummvm'
+scummvmSaves = SAVES + '/scummvm'
+
+viceBin = '/usr/bin/x64'
+viceConfig = CONF + "/vice/vice.conf"
+
+# Please keep this one at the end, and complete with your new binaries
+recalboxBins = [dosboxBin,
```
Comment (digitalLumberjack) [inline configgen/recalboxFiles.py]: should be a dict, that way we have no choice we must keep it up to date.

Comment (substring) [inline configgen/recalboxFiles.py]: ok. 
Maybe we should think about a way to gather systems in configgen. Now we have to declare them in both emulatorlauncher and recalboxFiles. There must be a more clever way. Maybe add some attributes to the Emulator class ?

**Comment by substring** [comment]
@digitalLumberjack PR updated. The dict is indexed on the name member of the emulator class rather than on the system. That's to manage multiple emulators for a single system

**Comment by digitalLumberjack** [comment]
Good point. 
The only thing that makes it not perfect is that there are two references to the same variable (one with recalboxBins, the other with recalboxFiles)

We should only have recalboxBins, and all the generators would have to use it.

**Comment by substring** [comment]
I totally agree. But it would mean to rewrite all generators to use `recalboxFiles.recalboxBin[system.emulator]` ... Want me to do that in this PR ?

**Comment by digitalLumberjack** [comment]
That goes with the dict :)

**Comment by substring** [comment]
updated

File: configgen/recalboxFiles.py [hunk 2]
```diff
@@ -68,7 +73,29 @@
...
+
+scummvmBin = '/usr/bin/scummvm'
+scummvmSaves = SAVES + '/scummvm'
+
+viceBin = '/usr/bin/x64'
+viceConfig = CONF + "/vice/vice.conf"
+
+# Please keep this one at the end, and complete with your new binaries
+recalboxBins = {"dosbox" : dosboxBin,
+				"fba2x" :fba2xBin,
```
Comment (digitalLumberjack) [inline configgen/recalboxFiles.py]: I would have set only one reference to the emulators values : 
"fba2x" : '/usr/bin/fba2x',

File: configgen/recalboxFiles.py [hunk 3]
```diff
@@ -23,11 +25,14 @@
 libretroExt = '_libretro.so'
 screenshotsDir = "/recalbox/share/screenshots/"
 savesDir = "/recalbox/share/saves/"
+
 fbaRoot = CONF + '/fba/'
 fbaCustom = fbaRoot + 'fba2x.cfg'
 fbaCustomOrigin = fbaRoot + 'fba2x.cfg.origin'
 fba2xBin = '/usr/bin/fba2x'
```
Comment (digitalLumberjack) [inline configgen/recalboxFiles.py]: And remove all those references

**Comment by substring** [comment]
which new comment will you find now :laughing:

**Pull request merged**